library/src/main/java/com/google/android/exoplayer/extractor/flv/FlvExtractor.java

```diff
@@ -0,0 +1,310 @@
+      long timestamp = tagHeaderBuffer.readUnsignedInt24();
+      timestamp = (tagHeaderBuffer.readUnsignedByte() << 24) | timestamp;
+      int streamId = tagHeaderBuffer.readUnsignedInt24();
+
+      currentTagHeader.type = type;
+      currentTagHeader.dataSize = dataSize;
+      currentTagHeader.timestamp = timestamp * 1000;
+      currentTagHeader.streamId = streamId;
+
+      // Sanity checks.
+      Assertions.checkState(type == TAG_TYPE_AUDIO || type == TAG_TYPE_VIDEO
+          || type == TAG_TYPE_SCRIPT_DATA);
+      Assertions.checkState(dataSize <= Integer.MAX_VALUE);
+      // Reuse tagData buffer to avoid lot of memory allocation (performance penalty).
+      if (tagData == null || dataSize > tagData.capacity()) {
+        tagData = new ParsableByteArray((int) dataSize);
+      } else {
+        tagData.setPosition(0);
+      }
+      tagData.setLimit(dataSize);
+      parserState = STATE_READING_SAMPLE;
+
+    } catch (EOFException eof) {
+      return false;
+    }
+
+    return true;
+  }
+
+  /**
+   * Reads payload of an FLV tag from the provided {@link ExtractorInput}.
+   * @param input The {@link ExtractorInput} from which to read.
+   * @param seekPosition
+   * @return One of {@link Extractor#RESULT_CONTINUE} and {@link Extractor#RESULT_END_OF_INPUT}.
+   * @throws IOException If an error occurred reading from the source.
+   * @throws InterruptedException If the thread was interrupted.
+   * @throws TagPayloadReader.UnsupportedTrack If payload of the tag is using a codec non
+   * supported codec.
+   */
+  private int readSample(ExtractorInput input, PositionHolder seekPosition) throws IOException,
+      InterruptedException, AudioTagPayloadReader.UnsupportedTrack {
+    if (tagData != null) {
+      if (!input.readFully(tagData.data, 0, currentTagHeader.dataSize, true)) {
+        return RESULT_END_OF_INPUT;
+      }
+      tagData.setPosition(0);
+    } else {
+      input.skipFully(currentTagHeader.dataSize);
+      return RESULT_CONTINUE;
+    }
+
+    // Pass payload to the right payload reader.
+    if (currentTagHeader.type == TAG_TYPE_AUDIO && audioReader != null) {
+      audioReader.consume(tagData, currentTagHeader.timestamp);
+    } else if (currentTagHeader.type == TAG_TYPE_VIDEO && videoReader != null) {
+      videoReader.consume(tagData, currentTagHeader.timestamp);
+    } else if (currentTagHeader.type == TAG_TYPE_SCRIPT_DATA && metadataReader != null) {
+      metadataReader.consume(tagData, currentTagHeader.timestamp);
+      if (metadataReader.durationUs != C.UNKNOWN_TIME_US) {
+        if (audioReader != null) {
+          audioReader.durationUs = metadataReader.durationUs;
+        }
+        if (videoReader != null) {
+          videoReader.durationUs = metadataReader.durationUs;
+        }
+      }
+    } else {
+      tagData.reset();
+    }
+
+    parserState = STATE_READING_TAG_HEADER;
+
+    return RESULT_CONTINUE;
+  }
+
+  // SeekMap implementation.
+  // TODO: Add seeking support
```

Comment:
It would be nice if this could be done in the initial commit, assuming the container provides sufficient information to allow for seeking. It's a pretty glaring absence :).